### ISSUE TITLE
--dump-config minimal dumps everything when a config file is given

### ISSUE THREAD

**Issue by emjotde** [marian-nmt/marian-dev#406]
@snukky What exactly is the expected behavior of `--dump-config minimal` with a given config yaml given as `--config file.yaml` ? It seems to dump everything that's given. I would expect to only dump non-default values?

**Comment by snukky** [comment]
This is correct. `--dump-config minimal` outputs all options that have been set, including options from a command line and config files.

We may add, for example, `--dump-config defaults`, which will be more aggressive and will ignore all command-line or config options that have a default value.

**Comment by emjotde** [comment]
"Minimal" seems the right name for the most aggressive option?
